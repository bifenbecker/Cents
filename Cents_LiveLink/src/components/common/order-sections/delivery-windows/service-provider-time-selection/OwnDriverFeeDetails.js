import React from "react";
import {Box, Flex, Image, Text} from "rebass/styled-components";

import {BlueVan} from "../../../../../assets/images";

import {toDollars} from "../../../../../utils/order-utils";

const OwnDriverFeeDetails = ({isPickup, deliveryFee}) => {
  const ownDriverDeliveryFee = deliveryFee?.ownDriver?.deliveryFeeInCents;

  return (
    <Flex {...styles.wrapper}>
      <Box>
        <Text {...styles.title}>
          When would you like your laundry to be {isPickup ? "picked up" : "delivered"}?
        </Text>
        <Text {...styles.deliveryFee}>
          {isPickup ? "Pickup fee: " : "Delivery fee: "}{" "}
          {deliveryFee ? toDollars(ownDriverDeliveryFee / 100) : "Free"}
        </Text>
      </Box>
      <Image
        sx={{transform: isPickup ? "none" : "rotateY(180deg)"}}
        src={BlueVan}
        {...styles.image}
      />
    </Flex>
  );
};

const styles = {
  wrapper: {
    justifyContent: "space-between",
    alignItems: "flex-start",
    fontSize: ["14px", "16px"],
  },
  title: {
    pb: "4px",
    width: "190px",
    sx: {
      overflowWrap: "break-word",
    },
  },
  deliveryFee: {
    fontSize: "24px",
    py: "8px",
  },
  subtext: {
    fontFamily: "secondary",
    fontStyle: "italic",
  },
  image: {
    flexShrink: 0,
  },
};

export default OwnDriverFeeDetails;
