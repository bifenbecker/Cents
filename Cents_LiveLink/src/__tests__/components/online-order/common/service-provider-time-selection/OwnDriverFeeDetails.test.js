import React from "react";
import {render, waitFor} from "@testing-library/react";
import OwnDriverFeeDetails from "../../../../../components/common/order-sections/delivery-windows/service-provider-time-selection/OwnDriverFeeDetails";

const renderOwnDriverFeeDetailsComponent = props => {
  const {getByText} = render(<OwnDriverFeeDetails {...props} />);

  return {getByText};
};

describe("OwnDriverFeeDetails Component", () => {
  const deliveryFee = {
    ownDriver: {deliveryFeeInCents: 500},
  };

  it("Test OwnDriverFeeDetails UI snapshot when isPickup is true", async () => {
    expect(
      render(<OwnDriverFeeDetails isPickup={true} deliveryFee={deliveryFee} />)
    ).toMatchSnapshot();
  });

  it("Test OwnDriverFeeDetails UI snapshot when isPickup is false", async () => {
    expect(
      render(<OwnDriverFeeDetails isPickup={false} deliveryFee={deliveryFee} />)
    ).toMatchSnapshot();
  });

  describe("Test OwnDriverFeeDetails Text ==> laundry to be", () => {
    it("should display text When would you like your laundry to be picked up", async () => {
      const {getByText} = renderOwnDriverFeeDetailsComponent({
        isPickup: true,
        deliveryFee: {deliveryFee},
      });

      const pickedUpLabel = await getByText(
        "When would you like your laundry to be picked up?"
      );

      await waitFor(() => expect(pickedUpLabel).toBeInTheDocument());
    });

    it("should display text When would you like your laundry to be delivered", async () => {
      const {getByText} = renderOwnDriverFeeDetailsComponent({
        isPickup: false,
        deliveryFee: {deliveryFee},
      });

      const pickedUpLabel = await getByText(
        "When would you like your laundry to be delivered?"
      );

      await waitFor(() => expect(pickedUpLabel).toBeInTheDocument());
    });
  });
});
