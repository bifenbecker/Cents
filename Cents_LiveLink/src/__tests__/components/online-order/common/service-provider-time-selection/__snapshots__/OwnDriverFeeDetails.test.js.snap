// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnDriverFeeDetails Component Test OwnDriverFeeDetails UI snapshot when isPickup is false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi gJZKis hzPtvw"
        font-size="14px,16px"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          <div
            class="sc-bczRLJ copdwV"
            width="190px"
          >
            When would you like your laundry to be 
            delivered
            ?
          </div>
          <div
            class="sc-bczRLJ gfAZjI"
            font-size="24px"
          >
            Delivery fee: 
             
            $5.00
          </div>
        </div>
        <img
          class="sc-bczRLJ kTaUgy"
          src="Blue_Van.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi gJZKis hzPtvw"
      font-size="14px,16px"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        <div
          class="sc-bczRLJ copdwV"
          width="190px"
        >
          When would you like your laundry to be 
          delivered
          ?
        </div>
        <div
          class="sc-bczRLJ gfAZjI"
          font-size="24px"
        >
          Delivery fee: 
           
          $5.00
        </div>
      </div>
      <img
        class="sc-bczRLJ kTaUgy"
        src="Blue_Van.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`OwnDriverFeeDetails Component Test OwnDriverFeeDetails UI snapshot when isPickup is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi gJZKis hzPtvw"
        font-size="14px,16px"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          <div
            class="sc-bczRLJ copdwV"
            width="190px"
          >
            When would you like your laundry to be 
            picked up
            ?
          </div>
          <div
            class="sc-bczRLJ gfAZjI"
            font-size="24px"
          >
            Pickup fee: 
             
            $5.00
          </div>
        </div>
        <img
          class="sc-bczRLJ iNuvfW"
          src="Blue_Van.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi gJZKis hzPtvw"
      font-size="14px,16px"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        <div
          class="sc-bczRLJ copdwV"
          width="190px"
        >
          When would you like your laundry to be 
          picked up
          ?
        </div>
        <div
          class="sc-bczRLJ gfAZjI"
          font-size="24px"
        >
          Pickup fee: 
           
          $5.00
        </div>
      </div>
      <img
        class="sc-bczRLJ iNuvfW"
        src="Blue_Van.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
